### DISCUSSION TITLE
Add `metals_status` to statusline without lua

### DISCUSSION THREAD

**Discussion by gnfisher** [scalameta/nvim-metals#236] [Q&A] [answered]
I've been unable to figure out how to do this with a vanilla statusline. I'm using vimscript config still and haven't had time to dive into the lua side of things yet. I can't figure out how to include metals status in my vanilla statusline (not using airline, etc). Would appreciate a nudge in the right direction!

**Comment by ckipp01** [comment, marked as answer]
Hey @gnfisher, you should be able to define a function that grabs the variable like this:

```vim
function! metals#status() abort
  return get(g:, 'metals_status', '')
endfunction
```

Then when you put together your statusline you should be able to just include `%{metals#status()}`. Keep in mind that by default `metals_status` is off, so if you're using it make sure to turn it on in your `init_options`.

**Reply by ckipp01** [reply]
Also if it's helpful at all, I recently stopped using a plugin for my statusline, and you can see how I put everything together here: https://github.com/ckipp01/dots/commit/972de624d4571a02b079ea8ee4b44f115a714562. Although mine is in Lua.

**Reply by gnfisher** [reply]
Thanks, this worked great. Those dotfiles are super useful. One of these weekends I need to make the jump into the brave new world of an all lua config...
